### DISCUSSION TITLE
Understanding AMM

### DISCUSSION THREAD

**Discussion by ghost** [adultmm/AdultMediaManager#12] [Q&A]
Hi,

First of all, thanks for this initiative!

I've just started using Kodi on my RPi4 which is connected to an external drive with adult content downloaded from FileJoker/K2S etc and torrents.

Some of my files have studio, date, pornstar and title in their filenames, but most are not scraped.
Other files are just like 150-8.mp4 - couldn't it just check the length of the movie and give me the opportunity to select the source?

Also, if I click on Rescrape on files that has failed scraping, there's that Blinking button option. In some cases, removing some text from the field, it will scrape. Why didn't it find it in the first place?

Thanks again - it will be great when I understand this :)

**Comment by adultmm** [comment]
AMM tries to detect the appropriate scraper from the full path of the video file. So if the site name (or the url) can be found in the file name or any parent directories the scraper will be selected. (There are some other matching techniques, but this is the most important.) 

The blinking button means this matching didn't succeed and any of the meta scrapers (Data18, ThePordDB) could not find the video.

If you clear some text from the searched title and then get some results, it means one of the selected meta scrapers can find the video by the truncated text.
